src/Concordium/Client/Runner.hs

```diff
@@ -3445,81 +3448,388 @@ processIdentityShowCmd action backend =
+    addPendingUpdateV1 :: (MonadFail m)
+        => PendingUpdate -- |The pending update.
+        -> PendingUpdates 'Types.ChainParametersV1 -- |The update queues.
+        -> m ( Either
+                 (PendingUpdates 'Types.ChainParametersV1)
+                 (Types.AccountAddress, Types.AccountIndex -> PendingUpdates 'Types.ChainParametersV1)
+             )
+    addPendingUpdateV1 PendingUpdate{..} updates =
+      case puEffect of
+        Queries.PUERootKeys hlKeys -> enqueueM pRootKeysUpdateQueue hlKeys
+        Queries.PUELevel1Keys l1Keys -> enqueueM pLevel1KeysUpdateQueue l1Keys
+        Queries.PUELevel2KeysV0 _ -> enqueueFail
+        Queries.PUELevel2KeysV1 auths -> enqueueM pLevel2KeysUpdateQueue auths
+        Queries.PUEProtocol pUpdate -> enqueueM pProtocolQueue pUpdate
+        Queries.PUEElectionDifficulty eDiff -> enqueueM pElectionDifficultyQueue eDiff
+        Queries.PUEEuroPerEnergy eRate -> enqueueM pEuroPerEnergyQueue eRate
+        Queries.PUEMicroCCDPerEuro eRate -> enqueueM pMicroGTUPerEuroQueue eRate
+        Queries.PUEFoundationAccount aAddr -> return . Right $
+            (aAddr, \ai -> over pFoundationAccountQueue (enqueue puEffectiveTime ai) updates)
+        Queries.PUEMintDistributionV0 _ -> enqueueFail
+        Queries.PUEMintDistributionV1 mDist -> enqueueM pMintDistributionQueue mDist
+        Queries.PUETransactionFeeDistribution tfDist -> enqueueM pTransactionFeeDistributionQueue tfDist
+        Queries.PUEGASRewards gRewards -> enqueueM pGASRewardsQueue gRewards
+        Queries.PUEPoolParametersV0 _ -> enqueueFail
+        Queries.PUEPoolParametersV1 pParams -> enqueueM pPoolParametersQueue pParams
+        Queries.PUEAddAnonymityRevoker arInfo -> enqueueM pAddAnonymityRevokerQueue arInfo
+        Queries.PUEAddIdentityProvider ipInfo -> enqueueM pAddIdentityProviderQueue ipInfo
+        Queries.PUECooldownParameters cdParams ->
+          return . Left $ over pCooldownParametersQueue (fmap $ enqueue puEffectiveTime cdParams) updates
+        Queries.PUETimeParameters tParams ->
+          return . Left $ over pTimeParametersQueue (fmap $ enqueue puEffectiveTime tParams) updates
+      where
+        enqueueFail = fail "Response included a V0 chain parameter update but expected only V1."
+        enqueueM :: (Monad m) => ASetter (PendingUpdates 'Types.ChainParametersV1) a (UpdateQueue e) (UpdateQueue e) -> e -> m (Either a b)
+        enqueueM l v = return . Left $ enqueue' puEffectiveTime updates l v
+
+    -- |Helper function to enqueue elements in a @PendingUpdates@ instance.
+    enqueue' :: Types.TransactionTime -- |The effective time for the update.
+              -> PendingUpdates cpv -- |The instance to enqueue the update to.
+              -> ASetter (PendingUpdates cpv) a (UpdateQueue e) (UpdateQueue e) -- |The lens corresponding to the queue to add the update.
+              -> e -- |The value to enqueue.
+              -> a
+    enqueue' pue ups l v = over l (enqueue pue v) ups
```

Comment:
I'm not sure if I should be using `enqueue` here. Should I just be adding it to the list which represents the queue, and should the list be sorted by the effective update time?

Reply:
Thanks for the fast review. So no reporting of invariant errors of the received stream is not sorted according to the effective update time?